### PR TITLE
AWS driver fixes

### PR DESCRIPTION
**Pull request by achaiah**
## Related Issue \ discussion
N/A

## Patch Description
- Fixes AWS credential access that uses a token - existing code does not take into consideration AWS credentials that use key/secret/token triplet to access resources.

- Fixes instantiation within VPC without AvailabilityZones. In these situations you must supply a SubnetID instead of AvailabilityZone.

- Extends EBS volume creation to include existing volumes (specified by `ebs_snapshot_id`). Note: In current code implementation EBS is attached but never mounted.

## Testing Instructions
Credential authentication fails when attempting to use key/secret/token (due to the obvious lack of ability to specify a token)

SubnetID is a required parameter in certain VPC configurations where Availability Zones are not configured.

### CODE REVIEW

File: clearml/automation/aws_driver.py [hunk 1]
```diff
@@ -47,29 +48,49 @@ def from_config(cls, config):
...
+                    }
+                }
+            ]
+            
+        if resource_conf.get("subnet_id", None):
+            launch_specification["SubnetId"] = resource_conf["subnet_id"]
+        elif resource_conf.get("availability_zone", None):
+            launch_specification["Placement"] = {"AvailabilityZone": resource_conf["availability_zone"]}
+        else:
+            raise Exception('subnet_id or availability_zone must to be specified in the config')
```
Comment (jkhenning) [inline clearml/automation/aws_driver.py:93]: Hi @achaiah, I'm not sure this is entirely correct, can you point me to the documentation for this?

**Comment by jkhenning** [comment]
@achaiah is this still relevant?

**Comment by achaiah** [comment]
Yes, this is still definitely an issue unless it has been fixed separately from this PR.

**Comment by jkhenning** [comment]
OK, do you have any reply for the question I raised in the code comment?

**Comment by achaiah** [comment]
I did reply to your question above. At this moment I don't have official AWS documentation beyond what the AWS API error message requested me to supply (i.e. SubnetId is required if no default AvailabilityZone is configured).

**Comment by jkhenning** [comment]
I'm sorry, I didn't see the reply...

**Pull request merged**